[Bug] [Android] Android app is crashed when tap to ListView Cell MenuItem

Hello,

I have a Xamarin.Forms app. 
Xamarin.Forms version = 4.4.0.991757

One of my pages has a ListView. Items in this ListView are rendered with a DataTemplate. DataTemplate has a ViewCell. ViewCell has MenuItems in ContextActions:

```
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:d="http://xamarin.com/schemas/2014/forms/design"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             x:Class="MyApp.Pages.MyPage">

  <ContentPage.Resources>
    <ResourceDictionary>

      <DataTemplate x:Key="MyInProgressWPTemplate">
        <ViewCell>
          <ViewCell.ContextActions>
            <MenuItem Command="{Binding Source={x:Reference WorkPackageListWrapper}, Path=BindingContext.CompleteCommand}" CommandParameter="{Binding .}"
               Text="Complete" />
            <MenuItem Command="{Binding Source={x:Reference WorkPackageListWrapper}, Path=BindingContext.RejectCommand}" CommandParameter="{Binding .}"
               Text="Reject" IsDestructive="{Binding IsMine}" />
          </ViewCell.ContextActions>

          <Grid>
          <!-- grid content -->
          </Grid>
          
        </ViewCell>
      </DataTemplate>

      <!-- other data templates -->
      
      <templateSelectors:WPItemTemplateSelector x:Key="TemplateSelector"
                                                MyInProgressWPTemplate="{StaticResource MyInProgressWPTemplate}"
                                                CompletedWPTemplate="{StaticResource NoOptionsWPTemplate}"
                                                OtherInProgressWPTemplate="{StaticResource NoOptionsWPTemplate}"
                                                NotStartedWPTemplate="{StaticResource NotStartedWPTemplate}" />

    </ResourceDictionary>
  </ContentPage.Resources>

  <ContentPage.Content>
    <StackLayout x:Name="WorkPackageListWrapper">

      <ListView 
            x:Name="workPackageList"
            VerticalOptions="FillAndExpand"
            IsPullToRefreshEnabled="true"
            IsRefreshing="{Binding IsBusy, Mode=OneWay}"
            RefreshCommand="{Binding RefreshListCommand}"
            IsGroupingEnabled="True"
            ItemsSource="{Binding GroupedList}"
            ItemTemplate="{StaticResource TemplateSelector}"
            HasUnevenRows="True"
            SeparatorVisibility="Default"
            GroupDisplayBinding="{Binding GroupName}"
            GroupHeaderTemplate="{StaticResource GroupTemplate}">
      </ListView>

    </StackLayout>
  </ContentPage.Content>
</ContentPage>
```

And, sometimes, when on Android I tap the menu item of a cell, the app is crashed.
On other platforms (iOS and UWP) I never saw this issue.

The app crash exception:

```
System.NullReferenceException: Object reference not set to an instance of an object
  at Xamarin.Forms.Platform.Android.CellAdapter.OnActionItemClickedImpl (Android.Views.IMenuItem item) [0x0000d] in <47887651c5f548d39078a5efb6ca1116>:0 
  at Xamarin.Forms.Platform.Android.CellAdapter.global::Android.Support.V7.View.ActionMode.ICallback.OnActionItemClicked (Android.Support.V7.View.ActionMode mode, Android.Views.IMenuItem item) [0x0000b] in <47887651c5f548d39078a5efb6ca1116>:0 
  at Android.Support.V7.View.ActionMode+ICallbackInvoker.n_OnActionItemClicked_Landroid_support_v7_view_ActionMode_Landroid_view_MenuItem_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_mode, System.IntPtr native_item) [0x00017] in <58dccd5333a748ca8e438a2c600c06d9>:0 
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.71(intptr,intptr,intptr,intptr)
```

Best Regards,
Aleksandrs

And when I debug the app from the Visual Studio, when the app is crashed, I have the following log records in the Visual Studio debug output window:

```
03-04 22:35:07.337 I/MonoDroid(28420): UNHANDLED EXCEPTION:
03-04 22:35:07.340 I/MonoDroid(28420): System.NullReferenceException: Object reference not set to an instance of an object.
03-04 22:35:07.340 I/MonoDroid(28420):   at Xamarin.Forms.Platform.Android.CellAdapter.OnActionItemClickedImpl (Android.Views.IMenuItem item) [0x00007] in D:\a\1\s\Xamarin.Forms.Platform.Android\CellAdapter.cs:266 
03-04 22:35:07.340 I/MonoDroid(28420):   at Xamarin.Forms.Platform.Android.CellAdapter.global::Android.Support.V7.View.ActionMode.ICallback.OnActionItemClicked (Android.Support.V7.View.ActionMode mode, Android.Views.IMenuItem item) [0x0000b] in D:\a\1\s\Xamarin.Forms.Platform.Android\CellAdapter.cs:95 
03-04 22:35:07.340 I/MonoDroid(28420):   at Android.Support.V7.View.ActionMode+ICallbackInvoker.n_OnActionItemClicked_Landroid_support_v7_view_ActionMode_Landroid_view_MenuItem_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_mode, System.IntPtr native_item) [0x00017] in <58dccd5333a748ca8e438a2c600c06d9>:0 
03-04 22:35:07.340 I/MonoDroid(28420):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.64(intptr,intptr,intptr,intptr)
03-04 22:35:07.353 W/zygote64(28420): JNI RegisterNativeMethods: attempt to register 0 native methods for android.runtime.JavaProxyThrowable
03-04 22:35:07.356 D/Mono    (28420): DllImport searching in: '__Internal' ('(null)').
03-04 22:35:07.357 D/Mono    (28420): Searching for 'java_interop_jnienv_throw'.
03-04 22:35:07.357 D/Mono    (28420): Probing 'java_interop_jnienv_throw'.
03-04 22:35:07.357 D/Mono    (28420): Found as 'java_interop_jnienv_throw'.
**System.NullReferenceException:** 'Object reference not set to an instance of an object.'


03-04 22:36:09.969 E/mono-rt (28420): [ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object.
03-04 22:36:09.969 E/mono-rt (28420):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.64(intptr,intptr,intptr,intptr)
03-04 22:36:09.969 E/mono-rt (28420):   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.64(intptr,intptr,intptr,intptr)
```

--- a/Xamarin.Forms.Platform.Android/CellAdapter.cs
+++ b/Xamarin.Forms.Platform.Android/CellAdapter.cs
@@ -265,6 +265,11 @@ namespace Xamarin.Forms.Platform.Android
 
 		void OnActionItemClickedImpl(IMenuItem item)
 		{
+			if (ActionModeContext == null)
+			{
+				return;
+			}
+
 			int index = item.ItemId;
 			IMenuItemController action = ActionModeContext.ContextActions[index];
 

